lib/events/events.go

```diff
@@ -144,16 +148,23 @@ func NewLogger() *Logger {
 
 func (l *Logger) Log(t EventType, data interface{}) {
 	l.mutex.Lock()
-	dl.Debugln("log", l.nextID, t, data)
-	l.nextID++
-	e := Event{
-		ID:   l.nextID,
-		Time: time.Now(),
-		Type: t,
-		Data: data,
-	}
+	dl.Debugln("log", l.nextGlobalID, t, data)
+	l.nextGlobalID++
+
+	now := time.Now()
+
 	for _, s := range l.subs {
 		if s.mask&t != 0 {
+			s.nextSubscriptionID++
+
+			e := Event{
```

Comment:
So you'd keep a separate map of Subscription -> Next Subscription ID on the logger? It seems to be that that would require a lot of unnecessary bookkeeping. 

I'd be tempted to pull the 

```go
if s.mask&t != 0 {
	s.nextSubscriptionID++

	e := Event{
		GlobalID:       l.nextGlobalID,
		SubscriptionID: s.nextSubscriptionID,
		Time:           now,
		Type:           t,
		Data:           data,
	}

	select {
	case s.events <- e:
	default:
		// if s.events is not ready, drop the event
	}
}
```

out into a separate `func (s *Subscription) publish(e Event), which could also encapsulate knowledge of the `mask` and usage of the `events` channel?